### PR TITLE
tools: add lambda-promtail missing errchecks

### PR DESCRIPTION
**Pull request by joanlopez**
**What this PR does / why we need it**:

While using [Lambda Promtail](https://grafana.com/docs/loki/latest/clients/lambda-promtail/) for a pet project, I realized there were some missing error checks (and other minor corrections; see comments inline) that do not help to understand what's happening in case something fails.

### CODE REVIEW

File: tools/lambda-promtail/lambda-promtail/main.go [hunk 1]
```diff
@@ -136,11 +137,11 @@ func handler(ctx context.Context, ev map[string]interface{}) error {
...
 
-	switch event.(type) {
+	switch evt := event.(type) {
 	case *events.S3Event:
-		return processS3Event(ctx, event.(*events.S3Event))
+		return processS3Event(ctx, evt)
 	case *events.CloudwatchLogsEvent:
-		return processCWEvent(ctx, event.(*events.CloudwatchLogsEvent))
+		return processCWEvent(ctx, evt)
 	}
```
Comment (joanlopez) [inline tools/lambda-promtail/lambda-promtail/main.go:145]: There's no need for extra type conversion on each call

File: tools/lambda-promtail/lambda-promtail/promtail.go [hunk 2]
```diff
@@ -45,8 +45,9 @@ func newBatch(ctx context.Context, entries ...entry) (*batch, error) {
 	}
 
 	for _, entry := range entries {
-		err := b.add(ctx, entry)
-		return b, err
```
[the remark this answers is not in the document]
Comment (kavirajk) [reply tools/lambda-promtail/lambda-promtail/promtail.go:49]: > as a newBatch caller I'd expect all the entries to be added into the batch

Looks reasonable to me.

**Review by kavirajk — APPROVED** [review]
LGTM. Thanks @joanlopez

**Comment by joanlopez** [comment]
Thanks @kavirajk! :)

**Pull request merged**